docs/src/docs/asciidoc/user/resilience.adoc

```diff
@@ -0,0 +1,189 @@
+are two ways to handle that
+
+* Expiration should take care of it. So entries should have an expiration set to their expected time validity.
+* We configure "too long" which will clear the cache
+
+== Clustering
+
+When using a distributed cache, your Ehcache client will connect to a remote Terracotta server (or multiple in case
+of stripping) which act as the authoritative tier.
+
+This server data can be replicated to other servers called mirrors. How your client will handle being cut from
+the cluster is defined in `ClusteringServiceConfiguration`.
+
+=== Timeouts
+
+If a client doesn't receive an answer in the configured time, it will timeout. This can happen for different reasons
+including a network cut, a long full GC or a server down.
+
+Three different types of timeouts have been defined.
+
+* Read: Operations reading data from the server
+* Mutative: Operations modifying data on the server
+* Lifecycle: Operations related to the lifecycle like creating and destroying a cache or simply connecting to the server
+
+By default, all are set to 5 seconds.
+
+=== Retries
+
+When a client timeouts, it might retry right away. That are two parameters for that:
+
+* Number of retries (default is 0)
+* Sleep time between retries (default is 0)
+
+Note that for each retry, the timeout will be enforced. So, far example, 3 retries with a timeout of 5 seconds and a sleep
+of 1 second will mean 20 seconds before returning.
+
+*Note to myself:* Should the retry be on the resilience strategy?
+
+=== In case of failure
+
+If we failed to get an answer, the cache [#Resilience Strategy] will kick in and return whatever it was configured to return.
+
+Resilience strategy is not used for lifecycle events. So in this case you will get a `StoreAccessException`.
+
+=== After a failure
+
+We will now consider a server with no mirror.
+
+The first thing that can happen is a server that has problem answering correctly all the time but that is still there.
+In this case, after 10 timeouts (configurable), the client will consider the server down to prevent waiting on requests.
+It will perform heartbeats in background to determine when the situation goes back to normal and reconnect if needed.
+Meanwhile, it will rely on the resilience strategy to answer.
+
+*Note:* M&M will trigger an alarm if a server is acting erratically.
+
+It is important to notice that the client won't fallback to the resilience strategy when the caching tier answers.
+It also means that the caching tier might not receive updates from the server and become out of sync.
+
+This is independent of the consistency configured and configurable. You can pick the following strategies:
+
+* Rely on tier forever, so even if the server is officially lost
+* Rely on tier on hiccups. The server will keep the caching tier until it declares the server lost for good
+* Don't rely on tier.
```

Comment:
Don't rely on tier?  Isn't this just strong consistency with a really short fuse on kicking clients?

Reply:
Yes. But it could be used for EVENTUAL as well.